Sources/Sentry/SentryHttpTransport.m

```diff
@@ -133,7 +136,11 @@ - (void)sendEnvelope:(SentryEnvelope *)envelope
     // thread, which could be the main thread.
     __weak SentryHttpTransport *weakSelf = self;
     [self.dispatchQueue dispatchAsyncWithBlock:^{
-        [weakSelf.fileManager storeEnvelope:envelopeToStore];
+        NSString *path = [weakSelf.fileManager storeEnvelope:envelopeToStore];
+        if (![[NSFileManager defaultManager] fileExistsAtPath:path]) {
```

Comment:
`m`: `storeEnvelope` already has the info when it fails to store the envelope

https://github.com/getsentry/sentry-cocoa/blob/24e07440d85b8d07ae4a98885d0f8f440ede339e/Sources/Sentry/SentryFileManager.m#L313-L328

 We could avoid the extra call to `fileExistsAtPath` by either returning `nil` or changing the methods return value to a boolean. The SDK doesn't use the path returned from the method except here.

Reply:
We use the path in tests. I've opted to return nil if we cannot save. That way the old functionality is preserved and we can still check whether the method failed from the call side.